src/Microsoft.Health.Dicom.Core/Features/Workitem/WorkitemService.Cancel.cs

```diff
@@ -0,0 +1,134 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Globalization;
+using System.Threading;
+using System.Threading.Tasks;
+using EnsureThat;
+using FellowOakDicom;
+using Microsoft.Extensions.Logging;
+using Microsoft.Health.Dicom.Core.Exceptions;
+using Microsoft.Health.Dicom.Core.Features.Store;
+using Microsoft.Health.Dicom.Core.Features.Store.Entries;
+using Microsoft.Health.Dicom.Core.Features.Workitem.Model;
+using Microsoft.Health.Dicom.Core.Messages.Workitem;
+
+namespace Microsoft.Health.Dicom.Core.Features.Workitem
+{
+    /// <summary>
+    /// Provides functionality to process the list of <see cref="IDicomInstanceEntry"/>.
+    /// </summary>
+    public partial class WorkitemService
+    {
+        public async Task<CancelWorkitemResponse> ProcessCancelAsync(DicomDataset dataset, string workitemInstanceUid, CancellationToken cancellationToken)
+        {
+            EnsureArg.IsNotNull(dataset, nameof(dataset));
+
+            var workitemMetadata = await _workitemOrchestrator
+                .GetWorkitemMetadataAsync(workitemInstanceUid, cancellationToken)
+                .ConfigureAwait(false);
+
+            // Get the state transition result
```

Comment:
There are a few things we need to do here before passing the dataset to the orchestrator:
- Get the current dataset from blob
- Set the Scheduled Procedure Step Modification DateTime to the current time
- change the state
- Use the incoming payload dataset to set the corresponding values in Progress Information Sequence, including setting Procedure Step Cancellation DateTime to the current time

Reply:
Done. Please review.